From 8dc10cc350dd7fa7e7a540d90b56c81126b46d3c Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Fri, 8 Feb 2019 22:00:16 +0100
Subject: [PATCH 4/5] drbd_transport_tcp: casts to char* from void* where
 needed.

sizeof(*(void*)p) is undefined in C. We might submit this patch
upstream.
---
 drbd/drbd_transport_tcp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drbd/drbd_transport_tcp.c b/drbd/drbd_transport_tcp.c
index 49c8068..25be149 100644
--- a/drbd/drbd_transport_tcp.c
+++ b/drbd/drbd_transport_tcp.c
@@ -288,7 +288,7 @@ static int _dtt_send(struct drbd_tcp_transport *tcp_transport, struct socket *so
 		if (rv < 0)
 			break;
 		sent += rv;
-		iov.iov_base += rv;
+		iov.iov_base = ((char*) iov.iov_base) + rv;
 		iov.iov_len  -= rv;
 	} while (sent < size);
 
@@ -328,7 +328,7 @@ static int dtt_recv(struct drbd_transport *transport, enum drbd_stream stream, v
 	} else if (flags & GROW_BUFFER) {
 		TR_ASSERT(transport, *buf == tcp_transport->rbuf[stream].base);
 		buffer = tcp_transport->rbuf[stream].pos;
-		TR_ASSERT(transport, (buffer - *buf) + size <= PAGE_SIZE);
+		TR_ASSERT(transport, ((char*) buffer - (char*) *buf) + size <= PAGE_SIZE);
 
 		rv = dtt_recv_short(socket, buffer, size, flags & ~GROW_BUFFER);
 	} else {
@@ -340,7 +340,7 @@ static int dtt_recv(struct drbd_transport *transport, enum drbd_stream stream, v
 	}
 
 	if (rv > 0)
-		tcp_transport->rbuf[stream].pos = buffer + rv;
+		tcp_transport->rbuf[stream].pos = ((char*) buffer) + rv;
 
 	return rv;
 }
-- 
2.17.0

